Sources/XcodeProj/Objects/Files/PBXGroup.swift

```diff
@@ -126,6 +128,27 @@ public extension PBXGroup {
             .first(where: { $0.name == name })
     }
 
+    /// Returns the file in the group with the given path, based on locations on-disk.
+    /// - Note: Paths function as-in Xcode; with respect to the group's `PBXSourceTree`.
+    /// Internally, identity is calculated based on normalized absolute paths for all files involved; i.e. actual
+    /// location in the file system.
+    ///
+    /// - Parameters:
+    ///   - path: a file path to search for.
+    /// - Returns: an existing reference to that file, or `nil` if not found
+    func file(with path: Path) -> PBXFileReference? {
+        func absolutePath(for fileRef: PBXFileElement) -> Path? {
+            try? fileRef.fullPath(sourceRoot: ".")
+        }
+        
+        guard let groupAbsPath = absolutePath(for: self) else { return nil }
+        let fileNormalizedAbsolutePath = (groupAbsPath + path).normalize()
+        return children.first {
+            guard let candidateAbsPath = absolutePath(for: $0) else { return false }
+            return candidateAbsPath.normalize() == fileNormalizedAbsolutePath
+        } as? PBXFileReference
```

Comment:
The algorithm has now quite importantly changed, and is much improved, but requires some feedback:

- I realised that the functionality already exists (`fullPath(sourceRoot:)`) to generate an absolute path for all files involved. Even passing `.` to represent "from current dir"
- Therefore a relative directory is not required by the caller: simply convert the argument path, and all group paths to normalized absolute paths, and search for that

This way, it's guaranteed that any kind of path (relative, absolute) works as input, since absolute paths are always generated internally as the file system "source of truth"

Reply:
I think using "." for the sourceRoot is wrong and will only work correctly for cases when the working dir is the same as project location. In other cases it will work in unpredictable way e.g. if you will have absolute file reference in project and a relative one which will resolve in incorrect way they will not be comparable. I'd advise to pass a real sourceRoot to the method.